backend/src/main/kotlin/cz/loono/backend/api/service/AccountService.kt

```diff
@@ -41,55 +79,53 @@ class AccountService(
-            logger.error(
-                "Tried to update User Auxiliary for uid: $uid but no such account exists. " +
-                    "The account should have been created by the interceptor."
-            )
-            throw IllegalStateException("Tried to update User Auxiliary for uid: $uid but no such account exists.")
+        var updatedAccount: Account = account
+        accountUpdate.nickname?.let {
+            updatedAccount = accountRepository.save(updatedAccount.copy(nickname = accountUpdate.nickname))
         }
-
-        val copy = UserAuxiliary(
-            nickname = aux.nickname ?: account.userAuxiliary.nickname,
-            preferredEmail = aux.preferredEmail ?: account.userAuxiliary.preferredEmail,
-            birthdate = aux.birthdate ?: account.userAuxiliary.birthdate,
-            sex = account.userAuxiliary.sex ?: aux.sex,
-            profileImageUrl = aux.profileImageUrl
-        )
-        return accountRepository.save(account.copy(userAuxiliary = copy))
+        accountUpdate.prefferedEmail?.let {
+            updatedAccount = accountRepository.save(updatedAccount.copy(preferredEmail = accountUpdate.prefferedEmail))
+        }
+        accountUpdate.leaderboardAnonymizationOptIn?.let {
+            updatedAccount =
+                accountRepository.save(updatedAccount.copy(leaderboardAnonymizationOptIn = accountUpdate.leaderboardAnonymizationOptIn))
+        }
+        accountUpdate.appointmentReminderEmailsOptIn?.let {
+            updatedAccount =
+                accountRepository.save(updatedAccount.copy(appointmentReminderEmailsOptIn = accountUpdate.appointmentReminderEmailsOptIn))
+        }
+        accountUpdate.newsletterOptIn?.let {
+            updatedAccount =
+                accountRepository.save(updatedAccount.copy(newsletterOptIn = accountUpdate.newsletterOptIn))
+        }
+        if (updatedAccount.profileImageUrl != accountUpdate.profileImageUrl) {
+            updatedAccount =
+                accountRepository.save(updatedAccount.copy(profileImageUrl = accountUpdate.profileImageUrl))
+        }
+        return transformToAccountDTO(updatedAccount)
     }
 
-    fun assembleUserDto(base: BasicUser, aux: UserAuxiliary): UserDto {
-        val account = accountRepository.findByUid(base.uid) ?: throw LoonoBackendException(
-            status = HttpStatus.NOT_FOUND,
-            errorCode = "404",
-            errorMessage = "The account not found."
-        )
-        return UserDto(
-            uid = base.uid,
-            email = base.email,
-            nickname = aux.nickname ?: account.userAuxiliary.nickname,
-            sex = aux.sex?.let(SexDto::valueOf),
-            birthdateMonth = aux.birthdate?.monthValue,
-            birthdateYear = aux.birthdate?.year,
-            preferredEmail = aux.preferredEmail,
-            profileImageUrl = aux.profileImageUrl ?: account.userAuxiliary.profileImageUrl
+    fun transformToAccountDTO(account: Account): AccountDto {
```

Comment:
extension instead of return?
```suggestion
    fun transformToAccountDTO(account: Account): AccountDto =
```